Ab/updates for new flow

abarciauskas-bgse opened the pull request:
This PR modified the user-metadata-form so that we update or create a Cumulus provider given the users bucket input. Removes job ids which were previously associated with DPS jobs. Adds a test but could add more tests.

chuckwondo commented on README.md:
````diff
@@ -38,6 +41,14 @@ Once the migration is complete, check configuration by starting the server. Star
+docker build . -t umf
+docker run --env-file .env -p 2998:2998 --entrypoint bash -it umf
+```
+
````
I think we need to add details to this section. The following are not obvious to me:

- Is this an _alternative_ to following the instructions above (in Getting Started)?
- If so, what's next, after opening a bash session in the container? What do I do in the terminal?
- What do I put in my `.env` file?
- Is the intent to have docker connect to postgres running outside of the container, on the host? If so, this definitely needs details (or at least a link to relavent instructions online)

chuckwondo reviewed (changes requested):
Hi Aimee. Let's definitely pair up to discuss how to reasonably deal with the regex stuff for collection files and granuleId extraction.

chuckwondo commented on README.md:
````diff
@@ -38,6 +41,14 @@ Once the migration is complete, check configuration by starting the server. Star
+## Running Docker locally
+
+```
+cp deployment/Dockerfile .
+docker build . -t umf
````
No need for copying the Docker file. The `cp` command can be eliminated and the docker command can be changed to:
```
docker build -f deployment/Dockerfile -t umf .
```
This also means that there's no need to add `Dockerfile` to `.gitignore` since no copying should be happening.

chuckwondo commented:
Although not part of the changes to the PR, I suggest changing the contents of `.ruby-version` to drop the `ruby-` suffix, leaving only the version number (e.g., only `2.7.5`, not `ruby-2.7.5`). Doing so allows use of `rbenv` in addition to `rvm`. The `ruby-` prefix is not acceptable to `rbenv`, but dropping the suffix works with both `rbenv` and `rvm`. Since I have `rbenv` installed, I'm forced to modify `.ruby-version` in order to install the correct version.

abarciauskas-bgse commented on lib/cumulus_api.rb:
```diff
@@ -1,46 +1,83 @@
   def self.cumulus_collection(collection_data)
+    if collection_data.filename_regex != nil
+      regex = collection_data.filename_regex
+      # Not sure what this should be
+      granuleIdExtraction = "^(#{collection_data.filename_regex}.+)$"
```
I see your points and they are good ones and I'm inclined to start by just permitting users to specify the filename prefix they want to use to both find files and extract granuleIds, to simplify this process. And we can add flexibility in how users do this discovery and granuleId extraction in the future, if it's requested.

chuckwondo commented on README.md:
```diff
@@ -17,26 +17,48 @@ Before moving into installing all the gems used, please make sure that you have
+bundle exec rails db:migrate
+# Wait for the migration to complete.
+# Source environment variables used to call Cumulus.
+source .env
```
Using `source` here won't work because the `.env` file does not `export` the vars. However, if we were to `export` the vars, then using `docker run --env-file .env ...` would complain (i.e., we cannot export them when using `.env` as a docker `env-file` because docker expects only lines of the form `name=value`, not `export name=value`, or comment lines).

However, this is unnecessary anyway, because we're using the `dotenv` gem, which automatically makes use of the `.env` file, so you can simply remove lines 43-44.

chuckwondo reviewed (changes requested):
In addition to my review comments, remember to drop the `ruby-` prefix in the `.ruby-version` file.

abarciauskas-bgse commented on app/jobs/create_draft_job.rb:
```diff
@@ -11,8 +11,11 @@ def perform(questionnaire)
     collection_info = questionnaire.collection_info
-    cumulus_response = CumulusApi.create_cumulus_collection(collection_info)
-    Rails.logger.info "Created Cumulus collection for #{questionnaire.uid}: #{cumulus_response}"
+    cumulus_provider_response = CumulusApi.create_cumulus_provider(collection_info)
```
totally agree

abarciauskas-bgse replied:
https://github.com/NASA-IMPACT/active-maap-sprint/issues/306

abarciauskas-bgse commented:
Thanks Chuck for the review - I have addressed most of the comments but I agree that the running with docker is not currently working, will try and get that working.

chuckwondo reviewed (approved):
Looks good! I'm happy to approve and have you sort out the docker stuff later, if you want to get this PR wrapped up.
